### PR TITLE
Septentrion as lib

### LINKED ISSUE #60: Officialize use as lib (for Django-North)
- Make it easy to be called from outside without too many idiosyncrasies
- Give knobs to allow customizing non-critical parts of the workflow (tracking migrations, displaying, etc)

### PR DESCRIPTION
**Pull request by ewjoachim**
Closes #60 

### Successful PR Checklist:
- [ ] Tests
- [ ] Documentation (optionally: [run spell checking](https://github.com/peopledoc/septentrion/blob/master/CONTRIBUTING.rst#build-the-documentation))
- [x] Had a good time contributing? (feel free to give some feedback)

### CODE REVIEW

**Comment by ewjoachim** [comment]
I suggest we don't overcomplicate this PR, just add missing tests and merge as-is. We'll make a 2nd pass, maybe more ambitious.

**Comment by nadege** [comment]
rebased on master to fix conflict & new tests that were using `from_cli`

**Comment by nadege** [comment]
rebased on master to fix conflict & new tests that were using `from_cli`  :)

**Comment by nadege** [comment]
I ran into some issues while adding tests and ended up going a bit further into the "septentrion as lib" path.

File: septentrion/migration.py [hunk 1]
```diff
@@ -60,7 +62,11 @@ def migrate(
                 ):
                     run_script(settings=settings, path=path)
                     logger.info("Saving operation in the database")
+                    # TODO: do this XOR callback ?
+                    # And if callback only, do not create the table ?
```
Comment (k4nar) [inline septentrion/migration.py]: Is it intended as a "WIP" TODO which should be fixed before the merge?

Comment (nadege) [reply septentrion/migration.py]: no, it's a thought I had while adding the feature that I didn't want to forget as I don't have an answer yet.

Comment (k4nar) [reply septentrion/migration.py]: Ok, maybe it's worth an issue then :) .

File: septentrion/core.py [hunk 2]
```diff
@@ -2,11 +2,28 @@
...
 
+logger = logging.getLogger(__name__)
+
+
+def initialize(**kwargs):
+    settings = configuration.Settings(**kwargs)
+
+    level = configuration.log_level(verbosity=settings.VERBOSITY)
+    logging.basicConfig(level=level)
+    logger.info("Verbosity level: %s", logging.getLevelName(level))
```
Comment (ewjoachim) [inline septentrion/core.py]: I'm all for create_table in core, but a library should not mess with logs. I think that part belongs in cli.

Comment (nadege) [reply septentrion/core.py]: ah yes you're right

**Comment by nadege** [comment]
moved logging setup back in cli module, removed comment and I'll write an issue instead.

File: tests/conftest.py [hunk 3]
```diff
@@ -21,3 +23,18 @@ def db():
...
+    def execute(response):
+        yield response
+
+    def _fake_db(response=None):
+        execute_mock = mocker.patch(
+            "septentrion.db.execute", return_value=execute(response)
+        )
+        return execute_mock
+
+    return _fake_db
```
Comment (ewjoachim) [inline tests/conftest.py]: Didn't notice it in the other PR, but I believe this fixture could be simplified:
```
# definition: 
@pytest.fixture()
def fake_db(mocker):
	yield mocker.patch(
        "septentrion.db.execute", return_value=execute(response)
    )
# usage when you don't need a return value:
def test_something(fake_db):
    ...
# usage when you need a return value:
def tests_somethong(fake_db):
    fake_db.return_value = [True]
    ...
```

I may completely miss the point, though...?

**Comment by nadege** [comment]
oh yes.. I can approve and you merge ?

**Comment by ewjoachim** [comment]
Or you can approve and merge :D :+1:

**Comment by ewjoachim** [comment]
@mgu and @samuelhamard are codeowners !

Note: @nadege do you want to be codeowner too ?

**Pull request merged**